docs/Doxyfile

```diff
@@ -1,10 +1,17 @@
 INPUT = docs/cdoc/ dist/c/include/qiskit.h dist/c/include/qiskit/
 
 OPTIMIZE_OUTPUT_FOR_C = YES
-EXTRACT_ALL = YES
 ENABLE_PREPROCESSING = YES
 MACRO_EXPANSION = YES
 
+WARNINGS = YES
+WARN_IF_UNDOCUMENTED = YES
+WARN_IF_DOC_ERROR = YES
+WARN_IF_UNCOMPLETE_DOC = YES
+WARN_NO_PARAM_DOC = YES
+WARN_IF_UNDOC_ENUM_VAL = YES
+WARN_AS_ERROR = FAIL_ON_WARNINGS_PRINT
```

Comment:
I don't have a 100% clear handle on what `EXTRACT_ALL` was doing, other than suppressing warnings.  I think we're sufficiently explicit in the documentation builder that `EXTRACT_ALL` wasn't causing any changes, at least none that I could spot.